src/go/rdctl/pkg/wsl/wsl_windows.go

```diff
@@ -21,7 +25,34 @@ type WSL interface {
 type WSLImpl struct{}
 
 func (wsl WSLImpl) UnregisterDistros() error {
-	return factoryreset.UnregisterWSL()
+	cmd := exec.Command("wsl", "--list", "--quiet")
+	cmd.SysProcAttr = &windows.SysProcAttr{CreationFlags: windows.CREATE_NO_WINDOW}
+	rawBytes, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error getting current WSLs: %w", err)
+	}
+	decoder := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewDecoder()
+	actualOutput, err := decoder.String(string(rawBytes))
+	if err != nil {
+		return fmt.Errorf("error getting current WSLs: %w", err)
+	}
+	actualOutput = strings.ReplaceAll(actualOutput, "\r", "")
+	wsls := strings.Split(actualOutput, "\n")
```

Comment:
Consider using [`bufio.Scanner`](https://pkg.go.dev/bufio#Scanner); though given that the list should be small and we're just moving code, not changing this is fine too.

Reply:
I think I will leave it as it is for now - I'd rather just get this code moved.